staging/src/k8s.io/client-go/discovery/restmapper.go

```diff
@@ -107,6 +108,9 @@ func NewRESTMapper(groupResources []*APIGroupResources, versionInterfaces meta.V
 
 				plural := gv.WithResource(resource.Name)
 				singular := gv.WithResource(resource.SingularName)
+				kind := gv.WithResource(strings.ToLower(resource.Kind))
+				// handle cases where a "Kind" is given in place of a resource
+				versionMapper.AddSpecific(gv.WithKind(resource.Kind), kind, kind, scope)
```

Comment:
@deads2k everything mapper-related seems to deal with GroupVersionResources.
The builder obtains [the mapping here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/builder.go#L603), and goes on to call KindFor in delegate restmappers.

This current approach works with the test from comment https://github.com/kubernetes/kubernetes/pull/59227#issuecomment-363173986

Another approach could be to [have a fallback here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/builder.go#L613) so that when an error is returned because a `Kind.group/name` was specified instead of `Resource.group/name`, a `meta.GroupVersionKind` is returned from a new method in the schema package (schema.ParseGroupKindArg?)

Reply:
We spoke in person.  Don't mess with the restmapper.  This is about how the resource.Build chooses to parse input.